pallets/protos/src/lib.rs

```diff
@@ -859,6 +859,24 @@ pub mod pallet {
 							return false
 						}
 					},
+					Some(Categories::Shards(param_script_info)) => {
+						if let Categories::Shards(stored_script_info) = &struct_proto.category {
+							
+							let implementing_diffs: Vec<_> = param_script_info.implementing.clone().into_iter().filter(|item| stored_script_info.implementing.contains(item)).collect();
+							let requiring_diffs: Vec<_> = param_script_info.requiring.clone().into_iter().filter(|item| stored_script_info.requiring.contains(item)).collect();
+
+							if !implementing_diffs.is_empty() || !requiring_diffs.is_empty() || (param_script_info.format == stored_script_info.format) || (param_script_info == stored_script_info)
+							{
+								return Self::filter_tags(tags, struct_proto)
+							}
+							else {
+								return false
+							}
+						} else {
+							// it should never go here
+							return false
```

Comment:
shouldn't we make `filter_category()` return a `Result<bool, _>` then instead of `bool`, for better readability and for consistency?

Reply:
and for correctness

Reply:
The function is only internal and it is called by `filter_proto`, which also returns a `bool `. So, if we change `filter_category` we would need to also change `filter_proto`.
I would leave it as it is now.